cmake/modules/VTA.cmake

```diff
@@ -73,15 +73,30 @@ elseif(PYTHON)
 
   # Cycle accurate simulator driver build
   if(USE_VTA_TSIM)
+    if(NOT DEFINED ENV{VERILATOR_INC_DIR})
```

Comment:
This logic is not working out (CI is failing). I think it is generally better to have single if block rather than using a nested approach. For example:

```
if (NOT...)
elseif (EXISTS...)
elseif (EXISTS...)
else()
```

I double checked the docker image and the Verilator include dir is in `/usr/local/share/verilator/include`...here is some proof:

```
$ docker run --rm -it tvm_verilator bash
root@e7d413bdbd90:/# which verilator
/usr/local/bin/verilator
root@e7d413bdbd90:/# ls /usr/local/share/verilator/include/
gtkwave                verilated_cov.cpp    verilated_heavy.h       verilated_syms.h         verilated_vcd_sc.cpp
verilated.cpp          verilated_cov.h      verilated_imp.h         verilated_threads.cpp    verilated_vcd_sc.h
verilated.h            verilated_cov_key.h  verilated_intrinsics.h  verilated_threads.h      verilated_vpi.cpp
verilated.mk           verilated_dpi.cpp    verilated_save.cpp      verilated_trace.h        verilated_vpi.h
verilated.v            verilated_dpi.h      verilated_save.h        verilated_trace_imp.cpp  verilatedos.h
verilated_config.h     verilated_fst_c.cpp  verilated_sc.h          verilated_vcd_c.cpp      vltstd
verilated_config.h.in  verilated_fst_c.h    verilated_sym_props.h   verilated_vcd_c.h
```

Reply:
sorry, I don't quite get it. I see several reasons CI build fails.
1. ci_arm. No "RUN bash /install/ubuntu_install_verilator.sh" in docker/Dockerfile.ci_arm. As a result "Verilator not found"
2. ci_i386. Verilator was found, but __STDC_FORMAT_MACROS is unconditionally redefined in Verilator source and causes compilation error with -Werror
3. ci_gpu. TSIM is enabled for some reason. Verilator is not installed in docker file. "Verilator not found."

So Cmake script looks working, but Verilator is not present in all scripts that build TSIM. And when it is present, tsim lib cannot compile because of Verilator source quality and -Werror requirement.
@vegaluisjose

Reply:
Fair enough. It will be still nice to rewrite that `if` logic for readability.

@tmoreau89 do you know what is the strategy (CI) for TSIM on gpu/arm images?

Reply:
1. Should arm, gpu, wasm configs build TSIM?
2. Is it acceptable to remove -Werror from tests/scripts/task_config_build_* files that build TSIM?
@vegaluisjose

Reply:
(1) I am not sure, CI has been in flux recently.
(2) I don't think it is possible but do you know if newer version of [Verilator](https://github.com/apache/tvm/blob/main/docker/install/ubuntu_install_verilator.sh#L24) might help?

Reply:
The latest tagged Verilator https://github.com/verilator/verilator/blob/v4.210/include/verilatedos.h#L279 still has it without check

Reply:
In Jenkinsfile I see only BUILD:CPU runnig tests/scripts/task_python_vta_tsim.sh. Does it mean that all other targets could have build tsim disabled by config.cmake set(USE_VTA_TSIM OFF)? @vegaluisjose

Reply:
yeah, that sounds right.

Reply:
Could you rebase? so we can re-run CI again?